src/SablierMerkleVCA.sol

```diff
@@ -82,61 +82,81 @@ contract SablierMerkleVCA is
     //////////////////////////////////////////////////////////////////////////*/
 
     /// @inheritdoc SablierMerkleBase
-    function _claim(uint256 index, address recipient, uint128 amount) internal override {
+    function _claim(uint256 index, address recipient, uint128 fullAmount) internal override {
         uint40 blockTimestamp = uint40(block.timestamp);
 
-        // Check: unlock start time is in the past.
-        if (_timestamp.start >= blockTimestamp) {
-            revert Errors.SablierMerkleVCA_ClaimNotStarted(_timestamp.start);
+        // Load the vesting schedule into memory to avoid multiple SLOADs.
+        MerkleVCA.Schedule memory vestingSchedule = _schedule;
+
+        // Check: schedule start time is in the past.
+        if (vestingSchedule.startTime >= blockTimestamp) {
+            revert Errors.SablierMerkleVCA_ClaimNotStarted(vestingSchedule.startTime);
         }
 
-        uint128 claimableAmount;
+        // Calculate the claim and foregone amount.
+        uint128 claimAmount;
+        uint128 forgoneAmount;
 
-        // Calculate the claimable amount.
-        if (_timestamp.end <= blockTimestamp) {
-            // If the unlock period has ended, the recipient can claim the full amount.
-            claimableAmount = amount;
+        if (vestingSchedule.endTime <= blockTimestamp) {
+            // If the vesting period has ended, the recipient can claim the maturity amount.
+            claimAmount = fullAmount;
         } else {
             // Otherwise, calculate the claimable amount based on the elapsed time.
-            uint40 elapsedTime = blockTimestamp - _timestamp.start;
-            uint40 totalDuration = _timestamp.end - _timestamp.start;
+            uint40 elapsedTime = blockTimestamp - vestingSchedule.startTime;
+            uint40 totalDuration = vestingSchedule.endTime - vestingSchedule.startTime;
 
-            // Safe to cast because the division results into a value less than `amount` which is already an `uint128`.
-            claimableAmount = uint128((uint256(amount) * elapsedTime) / totalDuration);
+            // Safe to cast because the division results into a value less than `fullAmount`, which is already an
+            // `uint128`.
+            claimAmount = uint128((uint256(fullAmount) * elapsedTime) / totalDuration);
 
             // Effect: update the forgone amount.
-            forgoneAmount += (amount - claimableAmount);
+            forgoneAmount = fullAmount - claimAmount;
+            totalForgoneAmount += forgoneAmount;
         }
 
         // Interaction: transfer the tokens to the recipient.
-        TOKEN.safeTransfer({ to: recipient, value: claimableAmount });
+        TOKEN.safeTransfer({ to: recipient, value: claimAmount });
 
         // Log the claim.
-        emit Claim(index, recipient, claimableAmount, amount);
+        emit Claim({ index: index, recipient: recipient, claimAmount: claimAmount, forgoneAmount: forgoneAmount });
```

Comment:
I think named args are very helpful when key value has different names. Since, here they are exactly same, isn't it clearer to just use the values?